src/features/security-cloud-game/SecurityCloudGame.js

```diff
@@ -1,82 +1,115 @@
 import PropTypes from 'prop-types';
-import React, { Component } from 'react';
-import { FormattedMessage } from 'react-intl';
-import throttle from 'lodash/throttle';
-import Draggable from 'react-draggable';
+import React, { useEffect, useLayoutEffect, useRef, useState } from 'react';
+import { FormattedMessage, injectIntl } from 'react-intl';
+
+import FocusTrap from '../../components/focus-trap';
 
 import DragCloud from './DragCloud';
 import DropCloud from './DropCloud';
 import messages from './messages';
+import { checkOverlap, getGridPosition, getRandomCloudPosition } from './utils';
 
 import './SecurityCloudGame.scss';
 
-class SecurityCloudGame extends Component {
-    static propTypes = {
-        /** Width/Height to set the drag and drop clouds to.  Defaults to 64 */
-        cloudSize: PropTypes.number,
-        /** Height to set the game to */
-        height: PropTypes.number.isRequired,
-        /** Function to call when the `DragCloud` is successfuly dropped onto the `DropCloud` */
-        onValidDrop: PropTypes.func,
-        /** Width to set the game to */
-        width: PropTypes.number.isRequired,
-    };
+// pick these numbers to balance accessibility and game complexity
+const CLOUD_SIZE_RATIO = 5;
+const GRID_TRACK_SIZE_RATIO = 20;
+
+const SecurityCloudGame = ({ height, intl, onValidDrop, width }) => {
+    const [cloudSize, setCloudSize] = useState(null);
+    const [gridTrackSize, setGridTrackSize] = useState(null);
+    const [dropCloudPosition, setDropCloudPosition] = useState(null);
+    const [dragCloudPosition, setDragCloudPosition] = useState(null);
+    const [gameBoardHeight, setGameBoardHeight] = useState(null);
+    const [gameState, setGameState] = useState({
+        isValidDrop: false,
+        isOverlap: false,
+        liveText: '',
+    });
+
+    const messageElementRef = useRef();
+    // to handle resize events
+    const prevHeight = useRef();
+    const prevWidth = useRef();
```

Comment:
> would we be able to use one useRef to track both values? I think it would be cleaner than tracking them separately.

Yes, I made the change to use only one ref.

> also, would we be able to pass this new value to DragCloud instead of `boardHeight` and `boardWidth`?

Done.